packages/sw-precaching/src/lib/revisioned-cache-manager.js

```diff
@@ -92,101 +128,132 @@ class RevisionedCacheManager {
     this._eventsRegistered = true;
 
     self.addEventListener('install', (event) => {
-      if (this._cachedAssets.length === 0) {
+      if (this._fileEntriesToCache.length === 0) {
         return;
       }
 
       event.waitUntil(
         this._performInstallStep()
-        .then(() => {
-          this._close();
-        })
       );
     });
   }
 
-  _performInstallStep() {
-    const assetsAndHahes = this._cachedAssets;
-    const cacheId = null;
-    if (!Array.isArray(assetsAndHahes)) {
-      throw ErrorFactory.createError('assets-not-an-array');
-    }
-
-    const cacheName = cacheId || defaultCacheName;
+  _performInstallStep(cacheName) {
+    cacheName = cacheName || defaultCacheName;
 
+    let openCache;
     return caches.open(cacheName)
-    .then((openCache) => {
-      const cachePromises = assetsAndHahes.map((assetAndHash) => {
-        return this._getRevisionDetails(assetAndHash.path)
-        .then((previousRevisionDetails) => {
-          if (previousRevisionDetails === assetAndHash.revision) {
-            return openCache.match(assetAndHash.path)
-            .then((result) => {
-              if (result) {
-                return true;
-              }
-              return false;
-            });
-          } else {
-            return false;
-          }
-        })
-        .then((isAlreadyCached) => {
-          if (isAlreadyCached) {
+    .then((oC) => {
```

Comment:
Can we expand `oC` to a more readable variable name? I know you need to assign to `openCache` but you could use `openedCache` or something similar.